File: online_judge/UVA/00151/00151_GPE.cpp
/*
Power Crisis

UVA 151：https://onlinejudge.org/index.php?option=com_onlinejudge&Itemid=8&page=show_problem&problem=87
ZeroJudge q891：https://zerojudge.tw/ShowProblem?problemid=q891
GPE 考古：21944
*/

#include <bits/stdc++.h>
using namespace std;

int main() {
    int n;
    while (cin >> n && n != 0) {
        for (int m = 1; ; m++) {
            // initialize the input
            vector<int> region(n);
            for (int i = 0; i < n; i++) {
                region[i] = i+1;
            }

            int shut_down_region = 0;
            region.erase(region.begin());

            while (region.size() > 1) {
                shut_down_region = (shut_down_region + m - 1) % region.size();
                if (region[shut_down_region] == 13) {
                    break;
                }

                region.erase(region.begin() + shut_down_region);
            }

            if (region.size() == 1) {
                cout << m << endl;
                break;
            }
        }
    }

    return 0;
}

File: online_judge/UVA/00151/00151_GPE.py
"""
Power Crisis

UVA 151: https://onlinejudge.org/index.php?option=com_onlinejudge&Itemid=8&page=show_problem&problem=87
ZeroJudge q891: https://zerojudge.tw/ShowProblem?problemid=q891
GPE 考古: 21944
"""

while True:
    num_region = int(input())
    if num_region == 0:
        break

    region_id = [ _ for _ in range(1, num_region+1)]

    for i in range(1, num_region+1):
        del region_id[0]
        delete_region = 0
        while len(region_id) > 1:
            delete_region = (delete_region+i-1) % len(region_id)
            if region_id[delete_region] == 13:
                break
            del region_id[delete_region]
        
        if len(region_id) == 1:
            print(f"{i}")
            break
        region_id = [ _ for _ in range(1, num_region+1)]